notify/notify.go

```diff
@@ -665,8 +672,9 @@ func (r RetryStage) Exec(ctx context.Context, l log.Logger, alerts ...*types.Ale
 			r.metrics.notificationLatencySeconds.WithLabelValues(r.integration.Name()).Observe(time.Since(now).Seconds())
 			r.metrics.numNotifications.WithLabelValues(r.integration.Name()).Inc()
 			if err != nil {
-				r.metrics.numFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
+				r.metrics.numTotalFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
 				if !retry {
+					r.metrics.numPermanentlyFailedNotifications.WithLabelValues(r.integration.Name()).Inc()
```

Comment:
the metric needs to be incremented before L701 too (when the context is done).

Reply:
you're right. Fixed, thanks.